### PR TITLE
'Last defined wins' style 'replace' merge mode.

### PR DESCRIPTION
**Pull request by far-blue**
Here's a suggested way to easily support replace-style merging of duplicates. The last defined version of a duplicated link wins. I'm making 2 assumptions here but I'm sure others can improve on my work if needed:
1) reading of link definitions from the multiple files is deterministic
2) the user does not want / need to mix merge styles within a merge set.

### CODE REVIEW

File: src/MergePlugin.php [hunk 1]
```diff
@@ -108,6 +108,16 @@ class MergePlugin implements PluginInterface, EventSubscriberInterface
     protected $recurse = true;
 
     /**
+     * Whether to replace duplicate links.
+	  *
+	  * Normally, duplicate links are resolved using Composer's resolver.
+	  * Setting this flag changes the behaviour to 'last definition wins'.
```
Comment (bd808) [inline src/MergePlugin.php]: Looks like you got some tabs in here instead of spaces.

File: tests/phpunit/MergePluginTest.php [hunk 2]
```diff
@@ -106,6 +106,42 @@ function ($args) use ($that) {
...
+        $dir = $this->fixtureDir(__FUNCTION__);
+
+        $root = $this->rootFromJson("{$dir}/composer.json");
+
+        $root->setRequires(Argument::type('array'))->will(
+            function ($args) use ($that) {
+                $requires = $args[0];
+                $that->assertEquals(2, count($requires));
+                $that->assertArrayHasKey('monolog/monolog', $requires);
+					 $that->assertEquals('1.10.0',
```
Comment (bd808) [inline tests/phpunit/MergePluginTest.php]: Tabs snuck in on you again.

**Comment by bd808** [comment]
> reading of link definitions from the multiple files is deterministic

I'm not sure that this is guaranteed to be true with the current implementation of `mergePackages()`. `glob()` is likely to be deterministic as long as the inode order of files doesn't change but that can't be guaranteed across multiple deployments or even time on a single deployment. We will probably need to add a stable sort of the results of each `glob()` call.

**Comment by far-blue** [comment]
Hopefully that's the tabs sorted :) We work to a tab-based coding standard so I have to remember to switch when working with other projects.

**Comment by far-blue** [comment]
Regarding the deterministic behaviour around what is 'earlier' and what is 'later' it really should have a direct relationship with the order of the config filenames in the include array. Where there is a glob that expands to multiple results then sorting alphabetically is sensible. Looking at the behaviour of the glob/array_map/array_reduce/array_merge combination in mergePackage() I believe that is the behaviour it current exhibits. glob() sorts alphabetically by default and produces indexed arrays while array_merge will simply append the arrays. I've not tested it yet - I'll throw together a unit test to check it (and prevent it changing unexpectedly in future) tomorrow :)

**Comment by far-blue** [comment]
Sorry for the delay. Had a busy week and then struggled initially to find a good way to test the correct order of the config file loading. I hope you all find the unit test I eventually came up with acceptable :)

**Comment by far-blue** [comment]
I feel this pull request has nothing left to finish off so I'd like it considered for merging, unless anyone else has thoughts.

**Pull request closed without merging**